File: 000_C# Training/C#_14_Interface/Java/MainClass.java
import java.util.Scanner;
import java.util.HashMap;

public class MainClass
{
    public static void main(String[] args)
    {
        HashMap<String, Vendor> vendors = new HashMap<String, Vendor>();

        vendors.put("Klockmann", new Klockmann());
        vendors.put("Bundespolizei", new Bundespolizei());
        
        for (String vendor : vendors.keySet())
            vendors.get(vendor).ReadPdf();

        new Scanner(System.in).nextLine();
    }
}

interface Vendor
{
    void ReadPdf();
    //void NextFunction();
}

class Klockmann implements Vendor
{
    public void ReadPdf()
    {
        System.out.println("Clock gun");
        HelperFunction();
    }

    public void HelperFunction()
    {
    }
}

class Bundespolizei implements Vendor
{
    public void ReadPdf()
    {
        System.out.println("Haende hoch");
    }
}

File: 000_C# Training/C#_14_Interface/C#/MainClass.cs
using System;
using System.Collections.Generic;

public class MainClass
{
    public static void Main()
    {
        Dictionary<string, Vendor> vendors = new Dictionary<string, Vendor>();

        vendors.Add("Klockmann", new Klockmann());
        vendors.Add("Bundespolizei", new Bundespolizei());
        
        foreach (string vendor in vendors.Keys)
            vendors[vendor].ReadPdf();

        Console.ReadLine();
    }
}

public interface Vendor
{
    void ReadPdf();
    //void NextFunction();
}

public class Klockmann : Vendor
{
    public void ReadPdf()
    {
        Console.WriteLine("Clock gun");
        HelperFunction();
    }

    public void HelperFunction()
    {
    }
}

public class Bundespolizei : Vendor
{
    public void ReadPdf()
    {
        Console.WriteLine("Haende hoch");
    }
}